### PR TITLE
Column numeric sorting issue fixes

### PR DESCRIPTION
**Pull request by hoffmanjs**
## Algorithms and processes gui does not sort columns with integers correctly 

Sorting integer columns incrementally or decreasing

## Solution

Describe your solution.

## how you tested the change

Start computations gui. Sort the algorithms numeric columns and on the processes tab sort the numeric columns.  Verify numeric sorting is correct.

## Where the following done:

- [ x] Tests. Check all that apply:
   - [ ] Unit tests created or modified that run during ant test.
   - [ ] Integration tests created or modified that run during integration testing
         (Formerly called regression tests.)
   - [ ] Test procedure descriptions for manual testing
- [ ] Was relevant documentation updated?
- [ ] Were relevant config element (e.g. XML data) updated as appropriate

If you aren't sure leave unchecked and we will help guide you to want needs changing where.

### CODE REVIEW

File: src/main/java/decodes/tsdb/compedit/ProcessesListPanel.java [hunk 1]
```diff
@@ -260,7 +260,8 @@ class ProcessesListPanelTableModel extends AbstractTableModel implements
 		CAPEdit.instance().compeditDescriptions
 		.getString("ProcessListPanel.TableColumn4") };
 	static int columnWidths[] = { 10, 25, 10, 55 };
-	static char sortType[] = { 'i', 'S', 'i', 'S' };
+//	static char sortType[] = { 'i', 'S', 'i', 'S' };
```
Comment (ktarbet) [inline src/main/java/decodes/tsdb/compedit/ProcessesListPanel.java:263]: we should remove the commented line.

File: src/main/java/decodes/tsdb/compedit/AlgoListTableModel.java [hunk 2]
```diff
@@ -51,7 +51,8 @@ public class AlgoListTableModel
 	ArrayList<AlgorithmInList> myList = new ArrayList<AlgorithmInList>();
 	private String columnNames[];
 	static int columnWidths[] = { 5, 20, 25, 8, 42};
```
Comment (ktarbet) [inline src/main/java/decodes/tsdb/compedit/AlgoListTableModel.java:53]:
```suggestion
	private static int columnWidths[] = { 5, 20, 25, 8, 42};
```

File: src/main/java/decodes/tsdb/compedit/AlgoListTableModel.java [hunk 3]
```diff
@@ -51,7 +51,8 @@ public class AlgoListTableModel
...
-	static char sortType[] = { 'i', 'S', 'S', 'i', 'S' };
+
+	static Class[] sortType = {Long.class, String.class, String.class, Integer.class, String.class};
```
Comment (ktarbet) [inline src/main/java/decodes/tsdb/compedit/AlgoListTableModel.java:55]:
```suggestion
	private static Class[] sortType = {Long.class, String.class, String.class, Integer.class, String.class};
```

File: src/main/java/decodes/tsdb/compedit/AlgoListTableModel.java [hunk 4]
```diff
@@ -51,7 +51,8 @@ public class AlgoListTableModel
 	ArrayList<AlgorithmInList> myList = new ArrayList<AlgorithmInList>();
```
Comment (ktarbet) [inline src/main/java/decodes/tsdb/compedit/AlgoListTableModel.java:51]:
```suggestion
	private ArrayList<AlgorithmInList> myList = new ArrayList<AlgorithmInList>();
```

File: src/main/java/decodes/tsdb/compedit/ProcessesListPanel.java (hunk 1 above)
Comment (ktarbet) [inline src/main/java/decodes/tsdb/compedit/ProcessesListPanel.java:263]:
```suggestion
```

**Comment by ktarbet** [comment]
Thanks @hoffmanjs  for this improvement.

I've built this branch for testing.  Sorting is working for me in Algorithms.

I'm not able to access the 'Proceses'  tab in the Computation Editor.    Instead I'm stuck on the 'Algorithms' tab after clicking on the 'processs' tab.  

I'm getting this error, from the log:

```

FAILURE 07/01/24 23:52:17 Exception in thread "AWT-EventQueue-0" java.lang.NullPointerException
FAILURE 07/01/24 23:52:17 	at decodes.tsdb.compedit.ProcessesListPanelTableModel.getColumnClass(ProcessesListPanel.java:289)
FAILURE 07/01/24 23:52:17 	at javax.swing.JTable.getColumnClass(Unknown Source)
FAILURE 07/01/24 23:52:17 	at javax.swing.JTable.getCellRenderer(Unknown Source)
FAILURE 07/01/24 23:52:17 	at javax.swing.plaf.basic.BasicTableUI.paintCell(Unknown Source)
FAILURE 07/01/24 23:52:17 	at javax.swing.plaf.basic.BasicTableUI.paintCells(Unknown Source)
FAILURE 07/01/24 23:52:17 	at javax.swing.plaf.basic.BasicTableUI.paint(Unknown Source)
FAILURE 07/01/24 23:52:17 	at javax.swing.plaf.ComponentUI.update(Unknown Source)
FAILURE 07/01/24 23:52:17 	at javax.swing.JComponent.paintComponent(Unknown Source)
FAILURE 07/01/24 23:52:17 	at javax.swing.JComponent.paint(Unknown Source)
FAILURE 07/01/24 23:52:17 	at decodes.gui.SortingListTable.paint(SortingListTable.java:142)
FAILURE 07/01/24 23:52:17 	at javax.swing.JComponent.paintChildren(Unknown Source)
FAILURE 07/01/24 23:52:17 	at javax.swing.JComponent.paint(Unknown Source)
FAILURE 07/01/24 23:52:17 	at javax.swing.JViewport.paint(Unknown Source)
FAILURE 07/01/24 23:52:17 	at javax.swing.JComponent.paintChildren(Unknown Source)
FAILURE 07/01/24 23:52:17 	at javax.swing.JComponent.paint(Unknown Source)
FAILURE 07/01/24 23:52:17 	at javax.swing.JComponent.paintChildren(Unknown Source)
FAILURE 07/01/24 23:52:17 	at javax.swing.JComponent.paint(Unknown Source)
FAILURE 07/01/24 23:52:17 	at javax.swing.JComponent.paintChildren(Unknown Source)
FAILURE 07/01/24 23:52:17 	at javax.swing.JComponent.paint(Unknown Source)
FAILURE 07/01/24 23:52:17 	at javax.swing.JComponent.paintChildren(Unknown Source)
FAILURE 07/01/24 23:52:17 	at javax.swing.JComponent.paint(Unknown Source)
FAILURE 07/01/24 23:52:17 	at javax.swing.JComponent.paintChildren(Unknown Source)
FAILURE 07/01/24 23:52:17 	at javax.swing.JComponent.paint(Unknown Source)
FAILURE 07/01/24 23:52:17 	at javax.swing.JComponent.paintChildren(Unknown Source)
FAILURE 07/01/24 23:52:17 	at javax.swing.JComponent.paint(Unknown Source)
FAILURE 07/01/24 23:52:17 	at javax.swing.JComponent.paintChildren(Unknown Source)
FAILURE 07/01/24 23:52:17 	at javax.swing.JComponent.paint(Unknown Source)
FAILURE 07/01/24 23:52:17 	at javax.swing.JComponent.paintToOffscreen(Unknown Source)
FAILURE 07/01/24 23:52:17 	at javax.swing.RepaintManager$PaintManager.paintDoubleBuffered(Unknown Source)
FAILURE 07/01/24 23:52:17 	at javax.swing.RepaintManager$PaintManager.paint(Unknown Source)
FAILURE 07/01/24 23:52:17 	at javax.swing.RepaintManager.paint(Unknown Source)
FAILURE 07/01/24 23:52:17 	at javax.swing.JComponent._paintImmediately(Unknown Source)
FAILURE 07/01/24 23:52:17 	at javax.swing.JComponent.paintImmediately(Unknown Source)
FAILURE 07/01/24 23:52:17 	at javax.swing.RepaintManager$4.run(Unknown Source)
FAILURE 07/01/24 23:52:17 	at javax.swing.RepaintManager$4.run(Unknown Source)
FAILURE 07/01/24 23:52:17 	at java.security.AccessController.doPrivileged(Native Method)
FAILURE 07/01/24 23:52:17 	at java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(Unknown Source)
FAILURE 07/01/24 23:52:17 	at javax.swing.RepaintManager.paintDirtyRegions(Unknown Source)
FAILURE 07/01/24 23:52:17 	at javax.swing.RepaintManager.paintDirtyRegions(Unknown Source)
FAILURE 07/01/24 23:52:17 	at javax.swing.RepaintManager.prePaintDirtyRegions(Unknown Source)
FAILURE 07/01/24 23:52:17 	at javax.swing.RepaintManager.access$1200(Unknown Source)
FAILURE 07/01/24 23:52:17 	at javax.swing.RepaintManager$ProcessingRunnable.run(Unknown Source)
FAILURE 07/01/24 23:52:17 	at java.awt.event.InvocationEvent.dispatch(Unknown Source)
FAILURE 07/01/24 23:52:17 	at java.awt.EventQueue.dispatchEventImpl(Unknown Source)
FAILURE 07/01/24 23:52:17 	at java.awt.EventQueue.access$500(Unknown Source)
FAILURE 07/01/24 23:52:17 	at java.awt.EventQueue$3.run(Unknown Source)
FAILURE 07/01/24 23:52:17 	at java.awt.EventQueue$3.run(Unknown Source)
FAILURE 07/01/24 23:52:17 	at java.security.AccessController.doPrivileged(Native Method)
FAILURE 07/01/24 23:52:17 	at java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(Unknown Source)
FAILURE 07/01/24 23:52:17 	at java.awt.EventQueue.dispatchEvent(Unknown Source)
FAILURE 07/01/24 23:52:17 	at java.awt.EventDispatchThread.pumpOneEventForFilters(Unknown Source)
FAILURE 07/01/24 23:52:17 	at java.awt.EventDispatchThread.pumpEventsForFilter(Unknown Source)
FAILURE 07/01/24 23:52:17 	at java.awt.EventDispatchThread.pumpEventsForHierarchy(Unknown Source)
FAILURE 07/01/24 23:52:17 	at java.awt.EventDispatchThread.pumpEvents(Unknown Source)
FAILURE 07/01/24 23:52:17 	at java.awt.EventDispatchThread.pumpEvents(Unknown Source)
FAILURE 07/01/24 23:52:17 	at java.awt.EventDispatchThread.run(Unknown Source)

**Comment by ktarbet** [comment]
@hoffmanjs  I was able to commit to your branch, this change below, and resolve the Null Pointer Exception on my computer.  It must have been specific to the database I was using (likely some null comments).

https://github.com/hoffmanjs/opendcs/commit/65786df11e0499beb149ffd8a5f3d6ead4004d2a

**Review by ktarbet — APPROVED** [review]

**Comment by ktarbet** [comment]
@MikeNeilson ,  we need your approval since I tinkered with this branch to fix a NPE .

**Review by MikeNeilson — APPROVED** [review]

**Pull request merged**